File: old_backend/important-core-code/go/vulnerable_command.go
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical command injection vulnerabilities for testing Aikido scanner

package main

import (
    "net/http"
    "os/exec"
)

func pingHandler(w http.ResponseWriter, r *http.Request) {
    host := r.URL.Query().Get("host")
    
    // VULNERABLE: Command injection via exec.Command with shell
    cmd := exec.Command("sh", "-c", "ping -c 4 "+host)
    output, _ := cmd.Output()
    w.Write(output)
}

func processHandler(w http.ResponseWriter, r *http.Request) {
    filename := r.URL.Query().Get("file")
    
    // VULNERABLE: Unsanitized input in command
    cmd := exec.Command("convert", filename, "output.png")
    cmd.Run()
}

func main() {
    http.HandleFunc("/ping", pingHandler)
    http.HandleFunc("/process", processHandler)
    http.ListenAndServe(":8080", nil)
}

File: old_backend/important-core-code/javascript/vulnerable-command.js
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical command injection vulnerabilities for testing Aikido scanner

const express = require('express');
const { exec, execSync } = require('child_process');
const app = express();

app.get('/ping', (req, res) => {
  const host = req.query.host;
  // VULNERABLE: Unsanitized input in shell command
  exec(`ping -c 4 ${host}`, (error, stdout) => {
    res.send(stdout);
  });
});

app.get('/convert', (req, res) => {
  const filename = req.query.file;
  // VULNERABLE: Command injection via filename
  const output = execSync(`convert ${filename} output.png`);
  res.send('Converted');
});

const spawn = require('child_process').spawn;
app.get('/list', (req, res) => {
  const dir = req.query.directory;
  // VULNERABLE: Even spawn with shell:true is dangerous
  const ls = spawn('ls', [dir], { shell: true });
  ls.stdout.on('data', (data) => res.write(data));
});

app.listen(3000);
